Automattic/newspack-newsletters
# Add/premium newsletters

### All Submissions:






### Changes proposed in this Pull Request:



Requires https://github.com/Automattic/newspack-newsletters/pull/1229 - let's wait for it to be reviewed

Implements the proof of concept to support Premium Lists tied to Woocommerce Membership plans

### How to test the changes in this Pull Request:

1. Checkout https://github.com/Automattic/newspack-plugin/pull/2611
2. Create a Subscription List via the wp-admin panel (that will create e Tag or Mailchimp group in the ESP)
3. Also enable a regular list
4. Activate Woocommerce Membership and create a Memebership plan
5. In the "Restrict content" tab, select "Subscription Lists" and add the List you just created as the restricted content, plus one other list
6. Add a Newsletters Subscription Form block to a page and select all available lists, including the premium ones
7. As an anonymous reader, confirm you don't see the premium lists in the block
8. Confirm that the registration block also does not display the premium lists
9. Perform the action required to be granted the membership (register or purchase a product)
10. Confirm that now you see the lists in the block
11. Confirm the in ESP dashboard that the user was automatically added to the lists when they were added to the plan
12. Verify your account and confirm you are able to manage your subscription to the premium lists in My Account > Newsletters
13. Cancel the membership and confirm the user was removed from the lists in the ESP dashboard
14. Try the same flow with different lists added to different membership plans at the same time
15. See pemhSX-ne-p2 for more scenarios. Think of other scenarios and see if they behave as you'd expect it


### Other information:
